### PR TITLE
Add site configuration flag to override ICC Authentication method

### PR DESCRIPTION
**Pull request by ivanvgh**
### Description:

This PR adds a site configuration flag to override the default ICC platform authentication.
Also remove unused code lines.

### Reviewers:

- [ ] @diegomillan
- [ ] @Squirrel18

### CODE REVIEW

File: openedx_external_enrollments/settings/common.py [hunk 1]
```diff
@@ -77,3 +77,4 @@ def plugin_settings(settings):
     settings.ICC_CREATE_USER_API_FUNCTION = "core_user_create_users"
     settings.ICC_LEARNER_ROLE_ID = "5"
     settings.ICC_HASH_LENGTH = 10
+    settings.ICC_AUTH_METHOD_OVERRIDE = "saml2"
```
Comment (diegomillan) [inline openedx_external_enrollments/settings/common.py:80]: It's not an override, it's the actual auth method to use that could be overwritten.
```suggestion
    settings.ICC_AUTH_METHOD = "saml2"
```

**Review by diegomillan — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py [hunk 2]
```diff
@@ -132,7 +132,10 @@ def _create_icc_user(self, data, duplicated_username):
                 'users[0][firstname]': user.first_name,
                 'users[0][lastname]': user.last_name,
                 'users[0][email]': user.email,
-                'users[0][auth]': 'saml2',
+                'users[0][auth]': configuration_helpers.get_value(
+                    'ICC_AUTH_METHOD_OVERRIDE',
+                    settings.ICC_AUTH_METHOD_OVERRIDE
```
Comment (Squirrel18) [inline openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:137]: Trailing comma. @ivanvgh

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py [hunk 3]
```diff
@@ -189,12 +192,8 @@ def _get_icc_user_from_xml_response(self, response, log_details, method_type):
...
             LOG.warning('User was not found. Proceeding to create it.')
         except Exception as error:  # pylint: disable=broad-except
-            exception_message = response['EXCEPTION']['MESSAGE'] if response.get('EXCEPTION') else error
-            log_details['response'] = {
-                'error': ('Failed to retrieve ICC user from response. Reason: %s' % exception_message),
-            }
-
-            LOG.error('Failed to retrieve ICC user from response. Reason: %s', exception_message)
+            LOG.error('Failed to retrieve ICC user from response. Reason: %s',
+                      response['EXCEPTION']['MESSAGE'] if response.get('EXCEPTION') else error)
```
Comment (Squirrel18) [inline openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:196]: Move the closing parenthesis to a new line. @ivanvgh

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py (hunk 2 above)
Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:137]: Done.

File: openedx_external_enrollments/external_enrollments/icc_external_enrollment.py (hunk 3 above)
Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/icc_external_enrollment.py:196]: Moved.

File: openedx_external_enrollments/settings/common.py (hunk 1 above)
Comment (ivanvgh) [reply openedx_external_enrollments/settings/common.py:80]: You are right, modified.

**Review by Squirrel18 — APPROVED** [review]

**Review by diegomillan — APPROVED** [review]

**Pull request merged**